duckduckgo/iOS
# error handling, improved icons and updated UI

Reviewer: Mia
Asana: 

https://app.asana.com/0/414235014887631/312386120368429
https://app.asana.com/0/413877547933097/407806711134826
https://app.asana.com/0/414235014887631/479951269529681
https://app.asana.com/0/414235014887631/477593094173837
https://app.asana.com/0/414235014887631/479951269529682

CC:

**Description**:

* Improved error handling, including navigation and UI.
* Applied red icons when things are 'bad' when protection is on.
* Applied indicator icons (should solve center-ing problem).
* Back navigation for error pages and update site rating correctly.

Note that I am waiting for assets for the detail screens, but there's only the encryption detail just now.

**Steps to test this PR**:

A grade site: DuckDuckGo.com 
D -> B grade site: EvansCycles.com
C grade site: example.com
B grade site: falkirkrpg.org.uk
C/D grade site (with full errors: cnn.com

Test 1 - https://app.asana.com/0/414235014887631/312386120368429
1. Visit the sites listed above and view the privacy protection.  
1. Observe that bad things will now have a red icon on the overview screen when protection is on.
1. Icons are grey when protection is off

Test 2 - https://app.asana.com/0/413877547933097/407806711134826
1. Visit the sites listed above and view the privacy protection.  
1. Observe grade indicators are now graphics (and should look centred, properly, etc)

Test 3 - https://app.asana.com/0/414235014887631/479951269529681
1. Perform a search for a D grade site (e.g. search for CNN)
1. Click search result, observe C/D grade
1. Click back, observe A grade for DuckDuckGo.com

Test 4 - https://app.asana.com/0/414235014887631/477593094173837 and https://app.asana.com/0/414235014887631/479951269529682
1. Visit a D grade site
1. Force an error condition (either go to http://example.url or use airplane mode)
1. Note Site grade is ?
1. Open privacy protection now shows an error page with same message
1. Close privacy protection
1. Navigate back to previous page, observe it doesn't change

Test 5 
1. Perform some tests on iPhone X. 
1. Open privacy protection.
1. Observe UI colour at bottom of the screen is correct and the animation does not flicker

Test 6
1. On an error screen open the menu from the address bar
1. Observe no bookmark or whitelist action

Test 7
1. Use airplane mode and then try to go to a valid site
1. Observe error screen
1. Use refresh from menu, nothing happens
1. Turn off airplane mode
1. Hit refresh
1. Observe site loads

###### Reviewer Checklist:


###### PR DRI Checklist:
